### ISSUE TITLE
MyBatteryMonitor.sh breaks on macOS Sequoia 15

### ISSUE THREAD

**Issue by daisuke-f** [daisuke-f/my-magic-mouse-battery-monitor#3]
Running a `MyBatteryMonitor.sh` on macOS Sequoia 15 shows "XPath set is empty" and battery values are not contained in the request message.

```
% ./MyBatteryMonitor.sh 
[2024-11-10 10:50:29] Start sending
XPath set is empty
XPath set is empty
[2024-11-10 10:50:29] Request: {"request":"sender data","data":[{"host":"mymac","key":"mouse.battery","value":""},{"host":"mymac","key":"keyboard.battery","value":""}]}
[2024-11-10 10:50:29] Response: 
% sw_vers
ProductName:		macOS
ProductVersion:		15.1
BuildVersion:		24B83
% 
```

**Comment by daisuke-f** [comment]
`ioreg -a -r -n AppleDeviceManagementHIDEventService` shows like this:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<array>
        <dict>
                ...
                <key>BatteryPercent</key>
                <integer>38</integer>
                ...
                <key>Product</key>
                <string>System Administrator’s Keyboard</string>
                ...
        </dict>
        <dict>
                ...
                <key>BatteryPercent</key>
                <integer>17</integer>
                ...
                <key>Product</key>
                <string>System Administrator’s Mouse</string>
                ...
        </dict>
</array>
</plist>
```

macOS Sonoma 14 and Sequoia 15 seem to have different product values.

|mac OS Sonoma 14|mac OS Sequoia 15              |
|----------------|-------------------------------|
|Magic Keyboard  |System Administrator’s Keyboard|
|Magic Mouse     |System Administrator’s Mouse   |

Maybe XPath can support both cases.

**Comment by daisuke-f** [comment]
```
% ioreg -a -r -n AppleDeviceManagementHIDEventService | xmllint --xpath '/plist/array/dict/key[text()="Product"]/following-sibling::*[1][text()="Magic Mouse" or text()="System Administrator’s Mouse"]/../key[text()="BatteryPercent"]/following-sibling::*[1]/text()' -
17
% ioreg -a -r -n AppleDeviceManagementHIDEventService | xmllint --xpath '/plist/array/dict/key[text()="Product"]/following-sibling::*[1][text()="Magic Keyboard" or text()="System Administrator’s Keyboard"]/../key[text()="BatteryPercent"]/following-sibling::*[1]/text()' -
37
%
```

**Comment by daisuke-f** [comment]
Fixed: 569f33a1615a4fd5c5212fdb22f6362ac7eec567
